linode/objwsconfig/framework_model.go

```diff
@@ -0,0 +1,53 @@
+package objwsconfig
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+// ResourceModel describes the Terraform resource rm model to match the
+// resource schema.
+type ResourceModel struct {
+	Bucket          types.String `tfsdk:"bucket"`
+	Cluster         types.String `tfsdk:"cluster"`
+	AccessKey       types.String `tfsdk:"access_key"`
+	SecretKey       types.String `tfsdk:"secret_key"`
+	IndexDocument   types.String `tfsdk:"index_document"`
+	ErrorDocument   types.String `tfsdk:"error_document"`
+	WebsiteEndpoint types.String `tfsdk:"website_endpoint"`
+}
+
+func (rm *ResourceModel) CopyFrom(other ResourceModel, preserveKnown bool) {
+	rm.Bucket = helper.KeepOrUpdateValue(rm.Bucket, other.Bucket, preserveKnown)
+	rm.Cluster = helper.KeepOrUpdateValue(rm.Cluster, other.Cluster, preserveKnown)
+	rm.AccessKey = helper.KeepOrUpdateValue(rm.AccessKey, other.AccessKey, preserveKnown)
+	rm.SecretKey = helper.KeepOrUpdateValue(rm.SecretKey, other.SecretKey, preserveKnown)
+	rm.IndexDocument = helper.KeepOrUpdateValue(rm.IndexDocument, other.IndexDocument, preserveKnown)
+	rm.ErrorDocument = helper.KeepOrUpdateValue(rm.ErrorDocument, other.ErrorDocument, preserveKnown)
+	rm.WebsiteEndpoint = helper.KeepOrUpdateValue(rm.WebsiteEndpoint, other.WebsiteEndpoint, preserveKnown)
+}
+
+func (rm *ResourceModel) ComputeFieldValue() {
+	rm.WebsiteEndpoint = types.StringValue(
+		fmt.Sprintf("%s.website-%s.linodeobjects.com", rm.Bucket.ValueString(), rm.Cluster.ValueString()),
```

Comment:
Could this be resolved dynamically from the cluster's `StaticSiteDomain` attribute similar to how we resolve the S3 endpoint in the [`ComputeS3Endpoint(...)` helper](https://github.com/linode/terraform-provider-linode/blob/dev/linode/helper/objects.go#L62)? It may require an additional API request but will future-proof us for hostname changes, etc.

Reply:
That's make sense, i updated the code, you can check it again.

Reply:
Thanks for the quick response! 

> BTW: what's your relase plan for the terraform provider? if this will get merged, i want to use this ASAP.

We generally release on a biweekly cadence with our next release being tomorrow. I don't think we'll quite be able to meet that deadline for this PR but I'll keep you updated if anything changes.